src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -42,6 +42,7 @@ public class HostController : Controller
         private readonly IScriptHostManager _scriptHostManager;
         private readonly IFunctionsSyncManager _functionsSyncManager;
         private readonly HostPerformanceManager _performanceManager;
+        private readonly SemaphoreSlim _drainModeSemaphore = new SemaphoreSlim(1, 1);
```

Comment:
This can't be class scoped, otherwise the locking will be specific to this instance and will not have the desired effect across API calls. In other words, this should be static.

Reply:
Good catch, fixed!